### PR TITLE
Added maximum Message size ServeOption

### PR DESCRIPTION
**Pull request by davidelienhard**
### Description of your changes

Fixes #

I have:

- [ ] Read and followed Crossplane's [contribution process].
- [ ] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Review by negz — APPROVED** [review]

File: sdk.go [hunk 1]
```diff
@@ -114,12 +116,22 @@ func Insecure(insecure bool) ServeOption {
 	}
 }
 
+// MaxMsgSize returns a ServeOption to set the max message size in bytes the server can receive.
```
Comment (negz) [inline sdk.go:119]: Ah, looks like this needs a slight tweak to appease the linter. 😄

Comment (davidelienhard) [reply sdk.go]: I modified the comment... :D

Comment (negz) [reply sdk.go]: Thank you! Sadly it looks like this needs another rebase, and DCO is failing again. ☹️ Sorry this small change has been such a pain to get in!

Comment (davidelienhard) [reply sdk.go]: Should have fixed it, or at least i hope! Sorry for the annoyment...

**Pull request merged**